network-api/networkapi/wagtailpages/rss.py

```diff
@@ -16,18 +17,27 @@ class RSSFeed(Feed):
     description = 'The Mozilla Foundation Blog'
 
     def items(self):
-        # Pull this object specifically using the English page title, as an IndexPage
-        # rather than a BlogIndexPage, to make sure we're not filtering out all the
-        # "featured" posts (which we need to do for site content purposes))
-        index = IndexPage.objects.get(title_en__iexact='Blog')
+        # Try to get the RSS items from cache first
+        feed_set = cache.get('rss_feed_set')
 
-        # If that doesn't yield the blog page, pull using the universal title
-        if index is None:
-            index = IndexPage.objects.get(title__iexact='Blog')
+        if feed_set is None:
+            # If we don't have an active cache, we build one: pull the index page
+            # specifically using the English page title, as an IndexPage rather than
+            # as a BlogIndexPage, to make sure we're not filtering out all the
+            # "featured" posts (which we need to do for site content purposes).
+            index = IndexPage.objects.get(title_en__iexact='Blog')
```

Comment:
With .get() if it doesn't find the index page it'll throw a `IndexPage.DoesNotExist` error. We'll want to `try/catch` this just in case the page titled "Blog" has its title changed in the future.